packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -76,10 +80,27 @@ const DateRangeCalendarArrowSwitcher = styled(PickersArrowSwitcher)({
 const DAY_RANGE_SIZE = 40;
 const weeksContainerHeight = (DAY_RANGE_SIZE + DAY_MARGIN * 2) * 6;
 
-const DayCalendarForRange = styled(DayCalendar)({
+const DayCalendarForRange = styled(DayCalendar)(({ theme }) => ({
   minWidth: 312,
   minHeight: weeksContainerHeight,
-}) as typeof DayCalendar;
+  [`&.${dateRangeCalendarClasses.dayDragging}`]: {
+    [`& .${dayClasses.dayOutsideRangeInterval}`]: {
+      '@media (pointer: fine)': {
+        '&:hover': {
+          border: `1px solid ${(theme.vars || theme).palette.grey[500]}`,
```

Comment:
I'm pretty sure that this style has been incorrectly migrated—it introduces a regression of not providing a grey border around the currently hovered day.
It should be applied inversely—in case the `dayDragging` is not active to avoid styles fighting beinging highlighting and hover. 😉

Reply:
The selector is wrong indeed, it is applied when dragging but it should be the opposite
Migrating from ownerState to CSS classes in a mental effort I don't like :laughing: